core/controllers/profile.py

```diff
@@ -400,11 +401,62 @@ class UrlHandler(base.BaseHandler):
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
+    def _get_logout_url(self, current_url):
+        """Prepares and returns logout url which will be handled by LogoutPage
+        handler.
+
+        Args:
+            current_url: str. The current url of the page.
+
+        Returns:
+            str. Logout URL to be handled by LogoutPage handler.
+        """
+
+        current_path = urlparse.urlsplit(current_url).path
+        redirect_url_on_logout = current_url
+
+        if current_path == '/preferences':
+            redirect_url_on_logout = '/'
+        elif current_path == feconf.LEARNER_DASHBOARD_URL:
+            redirect_url_on_logout = '/'
+        elif current_path == '/notifications_dashboard':
+            redirect_url_on_logout = '/'
+        elif current_path == feconf.CREATOR_DASHBOARD_URL:
+            redirect_url_on_logout = '/'
+        elif current_path == feconf.TOPICS_AND_SKILLS_DASHBOARD_URL:
+            redirect_url_on_logout = '/'
+        elif current_path.split('/')[1] == feconf.SKILL_EDITOR_URL_PREFIX[1:]:
+            redirect_url_on_logout = '/'
+        elif current_path.split('/')[1] == feconf.TOPIC_EDITOR_URL_PREFIX[1:]:
+            redirect_url_on_logout = '/'
+        elif current_path.split('/')[1] == feconf.STORY_EDITOR_URL_PREFIX[1:]:
+            redirect_url_on_logout = '/'
+        elif current_path.split('/')[1] == feconf.EDITOR_URL_PREFIX[1:]:
+            exploration_id = current_path.split('/')[2]
+            redirect_url_on_logout = (
+                '%s/%s' % (feconf.EDITOR_URL_PREFIX, exploration_id))
```

Comment:
I don't think this big block of `if`s is a good idea. From line 418 to line 433 it is basically the same case where we want to redirect to `/`. Also, now that I look at it most of it would be possible to do on frontend instead on the backend.

Reply:
Done. I removed the `if` block. But I didn't move this code to frontend as then I won't be able to use `feconf` to check for urls.